File: Data Structures & Algorithms/duplicate-integer/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums: number[]): boolean {
        const mp = new Set<number>();

        for (const num of nums) {
            if (mp.has(num)) return true;
            mp.add(num)
        }
        return false;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        for (int num : nums) {
            mp[num]++;
        }
        priority_queue<
            pair<int, int>,
            vector<pair<int, int>>,
            greater<pair<int, int>>
        > pq;
        for (auto& [num, freq] : mp) {
            pq.push({freq, num});
            if (pq.size() > k) {
                pq.pop();
            }

        }
        vector<int> result;
        while (!pq.empty()) {
            result.push_back(pq.top().second);
            pq.pop();
        }
        return result;
    }
};